docs/configuration/config-file-reference.md

```diff
@@ -5364,6 +5388,44 @@ time_window:
   # lookback delta) that the query should be within. If set to 0, it won't be
   # checked.
   [end: <int> | default = 0]
+
+# Query time range should be within this limit to match. If not set, it won't be
+# checked.
+time_range_limit:
+  # Query time range should be above or equal to this value to match. If set to
+  # 0, it won't be checked.
+  [min: <int> | default = 0]
+
+  # Query time range should be below or equal to this value to match. If set to
+  # 0, it won't be checked.
+  [max: <int> | default = 0]
```

Comment:
Can we document a bit the syntax here? i suspect its something like `-1d`, `1h` etc

Reply:
Is this config now overlapping with:

```
# Limit the query time range (end - start time of range query parameter and max
# - min of data fetched time range). This limit is enforced in the
# query-frontend and ruler (on the received query). 0 to disable.
# CLI flag: -store.max-query-length
[max_query_length: <duration> | default = 0s]
```

Should we deprecated this one?

Reply:
Same as: 

```
# Maximum duration into the future you can query. 0 to disable.
# CLI flag: -querier.max-query-into-future
[max_query_into_future: <duration> | default = 10m]
```

Reply:
updated time-range docs